depthmapXcli/runmethods.cpp

```diff
@@ -96,14 +97,95 @@ namespace dm_runmethods
         DO_TIMED("Writing graph", mgraph->write(cmdP.getOuputFile().c_str(),METAGRAPH_VERSION, false);)
     }
 
-    void linkGraph(const CommandLineParser &cmdP, const std::vector<Line> &mergeLines, IPerformanceSink &perfWriter)
+    void linkGraph(const CommandLineParser &cmdP, const LinkParser &parser, IPerformanceSink &perfWriter)
     {
+
         auto mgraph = loadGraph(cmdP.getFileName().c_str(), perfWriter);
-        SimpleTimer t;
-        PointMap& currentMap = mgraph->getDisplayedPointMap();
 
-        std::vector<PixelRefPair> newLinks = depthmapX::pixelateMergeLines(mergeLines, currentMap);
-        depthmapX::mergePixelPairs(newLinks, currentMap);
+        if (parser.getLinkMode() == LinkParser::LinkMode::UNLINK
+                && mgraph->getShapeGraphs().getDisplayedMap().getMapType() != ShapeMap::AXIALMAP) {
+            throw depthmapX::RuntimeException("Unlinking is only available for axial maps");
+        }
+
+        char delimiter = '\t';
+        std::stringstream linksStream;
+        if(!parser.getLinksFile().empty())
```

Comment:
I wonder whether all this code parsing inputs shouldn't really be in the parser? You pass the parser in anyway, so could all be done at parse time (and fail early if something's wrong) and then exposed on the parser.

Reply:
The way I understand the various cli parsers work is to just parse what the command line gives them for whichever mode they represent and not process any actual data. The data processing is done by the runmethods functionality which takes the "commands" from the parser and executes them.

If the user makes a mistake in the command line (say forgets an argument), then it's the parser that fails. If the user provides a file that's bad but all the command-line arguments were fine then the it's the process after the parser that should fail, even if that process involved parsing.